firewood/src/db.rs

```diff
@@ -231,20 +215,65 @@ impl Db {
         latest_rev_nodestore.check(opt)
     }
 
-    /// Propose a new batch that is processed in parallel.
+    /// Create a proposal with a specified parent. A proposal is created in parallel if `use_parallel`
+    /// is `Always` or if `use_parallel` is `BatchSize` and the batch is >= to the `BatchSize` value.
     ///
     /// # Panics
     ///
     /// Panics if the revision manager cannot create a thread pool.
-    pub fn propose_parallel(
+    #[fastrace::trace(name = "propose")]
+    fn propose_with_parent<F: Parentable>(
         &self,
         batch: impl IntoIterator<IntoIter: KeyValuePairIter>,
+        parent: &NodeStore<F, FileBacked>,
     ) -> Result<Proposal<'_>, api::Error> {
-        let parent = self.manager.current_revision();
-        let mut parallel_merkle = ParallelMerkle::default();
-        let immutable =
-            parallel_merkle.create_proposal(&parent, batch, self.manager.threadpool())?;
+        // If use_parallel is BatchSize, then perform parallel proposal creation if the batch
+        // size is >= BatchSize.
+        let batch = batch.into_iter();
+        let use_parallel = match self.use_parallel {
+            UseParallel::Never => false,
+            UseParallel::Always => true,
+            UseParallel::BatchSize(required_size) => batch.size_hint().0 >= required_size,
+        };
+        let immutable = if use_parallel {
+            let mut parallel_merkle = ParallelMerkle::default();
+            let span = fastrace::Span::enter_with_local_parent("parallel_merkle");
+            let immutable =
+                parallel_merkle.create_proposal(parent, batch, self.manager.threadpool())?;
+            drop(span);
+            immutable
```

Comment:
This seems more complicated than necessary. Why isn't this:
```suggestion
            parallel_merkle.create_proposal(parent, batch, self.manager.threadpool())?
```
since `span` goes out of scope and will be dropped automatically.